### ISSUE TITLE
implementing WarmupLinearSchedule schedulre

### ISSUE THREAD

**Issue by mhajiaghayi** [microsoft/DeepSpeed#339]
As adaptive learning rate with a decay has shown to be superior over fixed learning rate, may I suggest to implement WarmupLinearSchedule? To implement such scheduler, you would need to know the total-steps that can be calculated as a prod of of number of epochs and training data size. It has to be read in the python script.

thanks
